docs/user_guide/preemption.md

```diff
@@ -228,4 +228,31 @@ In this example, two imbalances are observed:
+
+#### Preemption Storm
+
+With setup like above, there is a side effect of increasing the possibilities of preemption storm or loop happening within the specific region between different state queues (siblings belonging to same parent).
+
+ReplicaSets are a good example to look at for looping and circular preemption. Each time a pod from a replica set is removed the ReplicaSet controller will create a new pod to make sure the set is complete. That auto-recreation could trigger loops as described below.
+
+![preemption_storm](../assets/preemption_storm.png)
+
+Replica set <i>State1 Repl</i> runs in queue <i>State1</i>. Replica set <i>State2 Repl</i> runs in the queue <i>State2</i>. Both queues belong to the same parent queue (they are siblings), <i>Country1</i>. The pods all run with the same settings for priority and preemption. There is no space left on the cluster. <i>State1</i> has no guaranteed quota, 4  pods of each vcores:1 are running and multiple pods of each vcores:1 of the replica set are pending. <i>State2</i> has no guaranteed quota, 4 pods of each vcores:1 are running and multiple pods of each vcores:1 of the replica set are pending. Both region, <i>region1</i> and country, <i>country1</i> queue usage is vcores:4. Since <i>region1</i> has a guaranteed quota of vcores:10 and usage of vcores:8 lower than its guaranteed quota leading to starvation of resources. All the queues (including both direct or indirect) below the parent queue are starving as it inherits the “under guaranteed” behavior from above said parent queue, <i>region1</i> calculation unless each state (leaf) queue has its own guaranteed quota. Now, either one of these state queues can trigger preemption. 
+
+Let's say, <i>state1</i> triggers preemption to meet resource requirements for pending pods.
+To make room for a <i>State1 Repl</i> pod, a pod from the <i>State2 Repl</i> set is preempted. Now, the pending <i>State1 Repl</i> pod moves from pending to running. Now, the next scheduling cycle comes. Let's say, <i>State2</i> triggers preemption to meet resource requirements for its pending pods. In addition to already existing pending pods, pod preempted (killed) in earlier scheduling cycles would have been recreated automatically by this time as it is a replica set. To make room for a <i>State2 Repl</i> pod, a pod from the <i>State1 Repl</i> set is preempted. Now, the pending <i>State2 Repl</i> pod moves from pending to running and preempted (killed) pod belonging to <i>State1 Repl</i> set would be recreated again. Now, the next scheduling cycle comes. Again, the whole loop repeats killing each other from the siblings without going anywhere leading to a preemption storm causing instability of the queues.
+
```

Comment:
Add an extension on this: it could even happen for a child queue below country 2 that gets caught in the preemption storm.